Add explanation for agglomerateByPrevalence

theotimeee opened the pull request:
This PR aims to solve the issue (https://github.com/microbiome/mia/issues/407) by making the function `agglomerateByPrevalence` easier to understand : 

- Specify the returned value
- Add details on the way the function works
- Clarify the examples

Commit:
add explanation for agglomerateByPrevalence in getPrevalence man page (#407)

TuomasBorman commented:
Very nice. I think we could still move it under agglomerate-methods with agglomerateByRank and agglomerateByVariable

I think this is more "agglomeration" than "prevalence" function, so that would be the best place for it

Commit:
Merge devel

Commit:
NEWS and description files update

Commit:
move agglomerateByPrevalence documentation to agglomerate-methods man page

Commit:
fix arguments and aliases bugs

Commit:
up

Commit:
correct aliases

TuomasBorman commented on R/getPrevalence.R:
```diff
@@ -594,23 +585,22 @@ setMethod("getPrevalentAbundance", signature = c(x = "SummarizedExperiment"),
 
-#' @rdname getPrevalence
-#' @aliases mergeFeaturesByPrevalence
+#' @rdname agglomerate-methods
+#' 
+#' @details
+#' \code{agglomerateByPrevalence} sums up the values of assays at the taxonomic 
+#' level specified by \code{rank} (by default the highest taxonomic level 
+#' available) and selects the summed results that exceed the given population 
+#' prevalence at the given detection level. The other summed values (below the 
+#' threshold) are agglomerated in an additional row taking the name indicated by
+#' \code{other_label} (by default "Other").
+#' 
```
Is it possible to add examples here instead of agglomerate.R? I am not sure if it works like that. Are the examples then included in agglomerate.methods.Rd along with examples from agglomerate.R?

theotimeee replied:
I moved it here and it looks like it works well.

TuomasBorman commented on R/agglomerate.R:
```diff
@@ -49,6 +49,9 @@
 #'   nested alternative experiments by default (default:
 #'   \code{strip_altexp = TRUE})
+#'   
+#' @param other_label A single \code{character} valued used as the label for the
+#'   summary of non-prevalent taxa. (default: \code{other_label = "Other"})
```
This could be added to getPrevalence.R above agglomerateByPrevalence function if possible.

TuomasBorman commented on R/agglomerate.R:
```diff
@@ -62,8 +65,11 @@
-#' @return A taxonomically-agglomerated, optionally-pruned object of the same
-#'   class as \code{x}.
+#' @return 
+#' \code{agglomerateByRank} returns a taxonomically-agglomerated, 
+#'   optionally-pruned object of the same class as \code{x}.
+#' \code{agglomerateByPrevalence} returns a taxonomically-agglomerated object
+#' of the same class as x and based on prevalent taxonomic results.
```
Also these things related to agglomerateByPrevalence could be moved if possible

Commit:
correct documentation of agglomerateByPrevalence

Commit:
up

TuomasBorman commented:
If you check the man page of agglomerate-methods, it does not describe detection and prevalence and as_relative parameters. (They are passed through with ...). You could add to agglomerate.R file those parameters under ... parameter.

" Passed to getPrevalence and getPrevalentTaxa functions. Used in agglomeratebyPrev function..." (Along with the current text.

As agglomerate.R already has ... parameter, you cannot ... above agglomerateByPrevalence function (otherwise it overwrites the one in agglomerate.R).

Commit:
merge devel

Merge branch 'devel' of https://github.com/microbiome/mia into agglomerateByPrevalence_doc

# Conflicts:
#	NEWS
#	R/deprecate.R
#	man/deprecate.Rd

Commit:
up

Commit:
add detection, prevalence and as.relative parameters to the agglomerate-methods man page

Commit:
correct parameters agglomerate-methods man page

Commit:
Merge devel

TuomasBorman commented:
See the man page. You could modify description so that it describes the agglomeration functions in more general (it now only says about agglomerateByRank) --> Agglomeration functions can be used to sum-up to rank or variable or prevalence...

Commit:
add general info on agglomeration functions

Commit:
merge devel

Merge branch 'devel' of https://github.com/microbiome/mia into agglomerateByPrevalence_doc

# Conflicts:
#	NEWS

TuomasBorman commented on NAMESPACE:
```diff
@@ -50,7 +50,6 @@ export(importQIIME2)
 export(importQZA)
 export(inner_join)
 export(left_join)
-export(loadFromBiom)
```
Always check. Now loadFromBiom is removed. That is because you forgot to add

```
#' @rdname deprecate
#' @export

```
above loadFromBiom in deprecate.R

Commit:
export loadFromBiom and 3 first generic functions in deprecate file

Commit:
merge devel

Merge branch 'devel' of https://github.com/microbiome/mia into agglomerateByPrevalence_doc

# Conflicts:
#	NEWS

theotimeee committed:
Update DESCRIPTION
